src/mbgl/map/vector_tile_data.cpp

```diff
@@ -95,39 +139,45 @@ bool VectorTileData::reparse(std::function<void()> callback) {
 }
 
 Bucket* VectorTileData::getBucket(const StyleLayer& layer) {
-    if (!isReady() || !layer.bucket) {
+    if (!layer.bucket) {
         return nullptr;
     }
 
-    return tileWorker.getBucket(layer);
+    const auto it = buckets.find(layer.bucket->name);
+    if (it == buckets.end()) {
+        return nullptr;
+    }
+
+    assert(it->second);
+    return it->second.get();
 }
 
 void VectorTileData::redoPlacement(float angle, float pitch, bool collisionDebug) {
-    if (angle == currentAngle &&
-        pitch == currentPitch &&
-        collisionDebug == currentCollisionDebug)
+    if (angle == currentAngle && pitch == currentPitch && collisionDebug == currentCollisionDebug)
+        return;
+
+    if (workRequest) {
```

Comment:
Shouldn't this check should go _after_ the assignments to `lastAngle`, etc, so that when the callback calls `redoPlacement` again it uses the most recent values?